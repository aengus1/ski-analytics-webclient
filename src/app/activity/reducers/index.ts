import {
  createSelector,
  createFeatureSelector,
  ActionReducerMap,
} from '@ngrx/store';

import * as fromActivities from './activity.reducer';
import * as fromFilters from './activity-filter.reducer';
import * as fromRoot from '../../reducers';
import * as fromLayout from '../../shared/layout/reducers/layout.reducer';


export interface ActivitiesState {
  activities: fromActivities.State;
  error: any;
}
export const reducers: ActionReducerMap<ActivitiesState> = {
  activities: fromActivities.reducer,
  error: null
};


export interface ActivityFiltersState {
  filters: fromFilters.State;
  error: any;
}
export const filterReducers: ActionReducerMap<ActivityFiltersState> = {
  filters: fromFilters.reducer,
  error: null
};


export interface State extends fromRoot.State {
  activities: ActivitiesState;
  filters: ActivityFiltersState;
}


/**
 * The createFeatureSelector function selects a piece of state from the root of the state object.
 * This is used for selecting feature states that are loaded eagerly or lazily.
 */
export const getActivitiesState = createFeatureSelector<ActivitiesState>('activities');

export const getActivityFiltersState = createFeatureSelector<ActivityFiltersState>('filters');

export const getLayoutState = createFeatureSelector<fromLayout.State>('layout');



/**
 * Every reducer module exports selector functions, however child reducers
 * have no knowledge of the overall state tree. To make them usable, we
 * need to make new selectors that wrap them.
 *
 * The createSelector function creates very efficient selectors that are memoized and
 * only recompute when arguments change. The created selectors can also be composed
 * together to select different pieces of state.
 */
export const getActivitiesEntitiesState = createSelector(
  getActivitiesState,
  state => state.activities);

export const getActivityFilterEntitiesState = createSelector(
  getActivityFiltersState,
  state => state.filters);

export const getSelectedActivityId = createSelector(
  getActivitiesEntitiesState,
  fromActivities.getSelectedActivityId
);

export const getActivitySport = createSelector(
  getActivitiesEntitiesState,
  fromActivities.getActivitySport
);

export const getActivitySubSport = createSelector(
  getActivitiesEntitiesState,
  fromActivities.getActivitySubSport
);

export const getSidebarContent = createSelector(
  getActivitiesEntitiesState,
  fromActivities.getSidebarContents
);


/**
 * Adapters created with @ngrx/entity generate
 * commonly used selector functions including
 * getting all ids in the record set, a dictionary
 * of the records by id, an array of records and
 * the total number of records. This reduces boilerplate
 * in selecting records from the entity state.
 */
export const {
  selectIds: getActivityIds,
  selectEntities: getActivityEntities,
  selectAll: getAllActivities,
  selectTotal: getTotalActivities,
} = fromActivities.adapter.getSelectors(getActivitiesEntitiesState);

export const getSelectedActivity = createSelector(
  getActivityEntities,
  getSelectedActivityId,
  (entities, selectedId) => {
    return selectedId && entities[selectedId];
  }
);


export const getUnfilteredActivity = createSelector(
  getActivitiesEntitiesState,
  fromActivities.getUnfilteredActivity
);


export const getShowSidebar = createSelector(
  getLayoutState,
  fromLayout.getShowSidebar
);


export const {
  selectIds: getActivityFilterIds,
  selectEntities: getActivityFilterEntities,
  selectAll: getAllActivityFilters,
  selectTotal: getTotalFilters
} = fromFilters.adapter.getSelectors(getActivityFilterEntitiesState);


export const selectActivityFilters = (state: ActivityFiltersState) => state.filters;




