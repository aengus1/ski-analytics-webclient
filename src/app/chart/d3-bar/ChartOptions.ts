export enum ChartOrientation {
  VERTICAL,
  HORIZONTAL
}


export class ChartOptions {

  constructor() {}

  public orientation: ChartOrientation = ChartOrientation.VERTICAL;
  public hasNumAxis = true;
  public hasNumLabel = false;

}
