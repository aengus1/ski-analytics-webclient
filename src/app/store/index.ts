import {
  ActionReducer,
  ActionReducerMap,
  createFeatureSelector,
  createSelector,
  MetaReducer
} from '@ngrx/store';
import * as activityReducer from '../activity/reducers';
import { environment } from '../../environments/environment';

export interface State {

}

export const reducers: ActionReducerMap<State> = {
 activity: activityReducer
};


export const metaReducers: MetaReducer<State>[] = !environment.production ? [] : [];
