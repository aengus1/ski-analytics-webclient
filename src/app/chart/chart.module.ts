import {NgModule} from '@angular/core';
import {CommonModule} from '@angular/common';
import {D3BoxplotComponent} from './components/d3-boxplot/d3-boxplot.component';
import {D3ChartComponent} from './components/d3-chart/d3chart.component';
import {D3Service} from 'd3-ng2-service';
import {D3RangeSliderComponent} from './components/d3-range-slider/d3-range-slider.component';
import {D3DualRangeSliderComponent} from './components/d3-dual-range-slider/d3-dual-range-slider.component';
import {D3BarComponent} from './d3-bar/d3-bar.component';
import {LoggerService} from '../shared/services/logger.service';
import {D3RadialComponent} from './components/d3-radial/d3-radial.component';

@NgModule({
  imports: [
    CommonModule
  ],
  declarations: [D3BoxplotComponent, D3ChartComponent, D3RangeSliderComponent, D3DualRangeSliderComponent,
    D3BarComponent, D3RadialComponent],
  providers: [D3Service, LoggerService],
  exports: [D3BoxplotComponent, D3RangeSliderComponent, D3DualRangeSliderComponent, D3BarComponent, D3RadialComponent]
})
export class ChartModule { }
