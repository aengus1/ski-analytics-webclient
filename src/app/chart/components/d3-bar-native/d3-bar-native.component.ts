import {
  ChangeDetectionStrategy,
  Component,
  ElementRef,
  Input,
  OnInit,
  ViewChild,
  ViewEncapsulation
} from '@angular/core';
import {D3ChartComponent} from '../d3-chart/d3chart.component';
import {D3Service} from 'd3-ng2-service';
import {LoggerService} from '../../../shared/services/logger.service';
import {ChartOptions} from './ChartOptions';
import {IntervalPipe} from '../../../shared/pipes/interval.pipe';


@Component({
  selector: 'app-d3-bar-native',
  template: `
    <svg width="[width]" height="[height]">
      <svg:g [attr.transform]="'translate('+ margin +','+ margin +')'">
        <svg:g #xaxis class="axis--x" [attr.transform]="'translate(0,'+height+')'"
               fill="none" font-size="10" font-family="sans-serif"
               text-anchor="middle" style="text-anchor; start">
          <svg:path class="domain" stroke="#000" [attr.d]="getXaxis()"></svg:path>
        </svg:g>
          <svg:g class="bar-holder">
            <svg:rect *ngFor="let bar of data; index as i;" class="bar"
                      [attr.width]="(width-margin-margin)/data.length"
                      [attr.height]="this.getHeight(bar)"
                      [attr.x]="this.xScale(this.xLabels[i])"
                      [attr.y]="this.yScale(bar)">
            </svg:rect>
          </svg:g>
          <svg:g class="bar-text">
            <svg:text *ngFor="let t of data; index as i;" class="bar-text"
                      [attr.width]="(width-margin-margin)/data.length"
                      [attr.height]="this.getHeight(t)"
                      [attr.x]="this.xScale(this.xLabels[i])"
                      [attr.y]="this.yScale(t) * 0.9"
                      style="display: inline; font-size: 0.6em;">
              {{this.intervalPipe.transform(t)}}
            </svg:text>
          </svg:g>
        <svg:g class="bar-text">
          <svg:text *ngFor="let t of data; index as i;" class="bar-text"
                    [attr.width]="(width-margin-margin)/data.length"
                    [attr.height]="this.getHeight(t)"
                    [attr.x]="this.xScale(this.xLabels[i])"
                    [attr.y]="height * 1.1"
                    style="display: inline; font-size: 0.6em;">
            {{this.xLabels[i]}}
          </svg:text>
        </svg:g>
        </svg:g>
        </svg>`,
  styleUrls: ['./d3-bar-native.component.css'],
  encapsulation: ViewEncapsulation.None,
  changeDetection: ChangeDetectionStrategy.OnPush
})
export class D3BarNativeComponent extends D3ChartComponent implements OnInit {

  @Input()
  width: number;

  @Input()
  height: number;

  @Input()
  data: number[];

  @Input()
  margin: number;

  @Input()
  private yLabel: string;

  @Input()
  private xLabels: string[];

  @Input()
  private chartOptions: ChartOptions;

  @ViewChild('xaxis') xaxisElement;

  private min: any;
  private max: any;
  protected logger: LoggerService;

  private maxY = null;
  private yScale = null;
  private xScale = null;


  constructor(element: ElementRef, d3Service: D3Service, logger: LoggerService, private intervalPipe: IntervalPipe) {
    super(element, d3Service, logger);
    this.logger = logger;
    this.intervalPipe.setLeadingZeroes(false);
    this.margin = 15;
    // this.maxY = this.d3.max(this.data);
    // this.yScale = this.d3.scaleLinear().domain([0, this.maxY]).range([this.height, 0]).nice()
    // : this.d3.scaleLinear().domain([0, this.maxY])
    //   .range([0, this.width]).nice();

  }

  ngOnInit() {
    this.maxY = this.d3.max(this.data);
    this.yScale = this.d3.scaleLinear().domain([0, this.maxY]).range([this.height, 0]).nice();
    this.xScale = this.d3.scaleBand().domain(this.xLabels).range([0, this.width]).paddingInner(0);

    // : this.d3.scaleBand().domain(this.xLabels)
    //   .range([0, height]).paddingInner(this.chartOptions.barSpacing);
  }

  getXaxis(): string {
    this.d3.axisBottom(this.xScale).tickSize(0)
    return 'M0,0H' + this.width;
  }

  public createChart() {
    if (this.width > 0) {
      // this.margin = {
      //   top: this.height * 0.1,
      //   right: this.width * 0.1,
      //   bottom: this.height * 0.2,
      //   left: this.chartOptions.orientation === ChartOrientation.VERTICAL ? this.width * 0.1 : this.width * 0.4
      // };
    }
    this.min = Infinity;
    this.max = -Infinity;

    // set up the svg element and the chart
    if (this.parentNativeElement !== null) {
      this.d3Svg.attr('width', this.width);
      this.d3Svg.attr('height', this.height);
    }
  }


  public getHeight(item: number): number {
    return this.height - this.yScale(item);
  }

  public getY(item: number): number {
    return 0;
  }


  public updateChart() {
    this.logger.info('[D3BarComponent] update chart triggered');
    // base
    if (!this.data) {
      return;
    }
    this.data = this.data.filter(d => (d !== -999 && !isNaN(d)));
    // const sel = this.d3Svg.selectAll('svg')
    //   .data([this.data])
    //   .enter()
    //   .append('g')
    //   .attr('transform', 'translate(' + this.margin.left + ',' + this.margin.top + ')');
    //
    // this.build(<any>sel);

  }

  // protected build(g: Selection<any, SVGSVGElement, any, any>) {
  //
  //   const svg = this.d3Svg.append('g').attr('transform', 'translate(' + this.margin.left + ', ' + this.margin.top + ')');
  //   // const width = this.width - this.margin.left - this.margin.right;
  //   // const height = this.height - this.margin.top - this.margin.bottom;
  //   const xScale = this.chartOptions.orientation === ChartOrientation.VERTICAL ?
  //     this.d3.scaleBand().domain(this.xLabels)
  //       .range([0, width]).paddingInner(this.chartOptions.barSpacing)
  //     : this.d3.scaleBand().domain(this.xLabels)
  //       .range([0, height]).paddingInner(this.chartOptions.barSpacing);
  //
  //   const bandwidth = xScale.bandwidth();
  //   const maxY = this.d3.max(this.data);
  //
  //   const yScale = this.chartOptions.orientation === ChartOrientation.VERTICAL ?
  //     this.d3.scaleLinear().domain([0, maxY])
  //       .range([height, 0]).nice()
  //     : this.d3.scaleLinear().domain([0, maxY])
  //       .range([0, width]).nice();
  //
  //   const xAxis = this.chartOptions.orientation === ChartOrientation.VERTICAL ? this.d3.axisBottom(xScale).tickSize(0) :
  //     this.d3.axisLeft(xScale).tickSize(0);
  //   const yAxis = this.chartOptions.orientation === ChartOrientation.VERTICAL ? this.d3.axisLeft(yScale).ticks(6).tickPadding(50) :
  //     this.d3.axisBottom(yScale).ticks(4);
  //
  //   if (this.chartOptions.orientation === ChartOrientation.VERTICAL) {
  //     svg.append('g').classed('axis--x', true)
  //       .style('text-anchor', 'start')
  //       .attr('transform', 'translate(0,' + height + ')')
  //       .call(xAxis);
  //   } else {
  //     svg.append('g').classed('axis--x', true)
  //       .style('text-anchor', 'start')
  //       .attr('transform', 'translate(-15,0)')
  //       .call(xAxis);
  //   }
  //
  //
  //   const barHolder = svg.append('g').classed('bar-holder', true);
  //
  //   const bars = barHolder.selectAll('rect.bar');
  //
  //   // .attr('x', this.chartOptions.orientation === ChartOrientation.VERTICAL ? (d, i) => xScale(this.xLabels[i]) : (d) => 0)
  //   // .attr('width', this.chartOptions.orientation === ChartOrientation.VERTICAL ? bandwidth : (d) => yScale(d))
  //   // .attr('y', (d) => yScale(d))
  //   // .attr('height', (d) => height - yScale(d));
  //
  //   if (this.chartOptions.orientation === ChartOrientation.VERTICAL) {
  //     bars.data(this.data)
  //       .enter().append('rect').classed('bar', true)
  //     // .classed('selected_bar', (d, i) => i === 1)  TODO -> add input for 'selected' bars and change color
  //       .attr('width', bandwidth)
  //       .attr('height', (d) => height - yScale(d))
  //       .attr('x', (d, i) => xScale(this.xLabels[i]))
  //       .attr('y', (d) => yScale(d));
  //   } else {
  //     bars.data(this.data)
  //       .enter().append('rect').classed('bar', true)
  //       .attr('width', (d) => yScale(d))
  //       .attr('height', bandwidth)
  //       .attr('x', 0)
  //       .attr('y', (d, i) => xScale(this.xLabels[i]));
  //   }
  //
  //   if (this.chartOptions.hasNumLabel) {
  //     const label = svg.append('g').classed('bar-text', true).selectAll('text').data(this.data).enter()
  //       .append('text').classed('barlabel', true)
  //       // .style('text-anchor', 'middle')
  //       // .attr('fill', 'black')
  //       // .attr('font-size', 10)
  //       .attr('x', this.chartOptions.orientation === ChartOrientation.VERTICAL
  //         ? (d, i) => bandwidth * i : (dl) => width / 2) // (d) => yScale(d))
  //       .attr('y', this.chartOptions.orientation === ChartOrientation.VERTICAL ?
  //         (d) => yScale(d) : (d, i) => bandwidth * i)
  //       .attr('dx', this.chartOptions.orientation === ChartOrientation.VERTICAL ? bandwidth / 2 : 0)
  //       .attr('dy', this.chartOptions.orientation === ChartOrientation.VERTICAL ? '-0.2em' : bandwidth / 2);
  //     if (this.chartOptions.hideZeroLabels) {
  //       label.style('display', (d: number) => (d === 0) ? 'none' : 'inline');
  //     }
  //     label.text((d: number) => this.chartOptions.yLabelFormat === YLabelFormat.NUMERIC ? d : this.intervalPipe.transform(d));
  //
  //   }
  //
  //   if (this.chartOptions.orientation === ChartOrientation.VERTICAL && this.chartOptions.hasNumAxis) {
  //     const yAxisEle = svg.append('g').classed('axis--y', true).call(yAxis);
  //
  //     yAxisEle.append('text').classed('barlabel', true)
  //       .attr('transform', 'rotate(-90) translate(' + height / 2 + ',0)')
  //       // .style('text-anchor', 'middle')
  //       // .style('fill', 'black')
  //       // .style('font-size', 10)
  //       .attr('dy', '-2.8em')
  //       .attr('y', this.margin.top);
  //   } else if (this.chartOptions.orientation === ChartOrientation.HORIZONTAL && this.chartOptions.hasNumAxis) {
  //     const yAxisEle = svg.append('g').classed('axis--y', true).call(yAxis);
  //
  //     yAxisEle.append('text').classed('barlabel', true)
  //     // .attr('transform', ' translate(' + height + ',0)')
  //       .style('text-anchor', 'right')
  //       //   .style('fill', 'black')
  //       //   .style('font-size', 10)
  //       .attr('dy', '-15em')
  //       .attr('y', 100);
  //   }
  //
  //
  // }
  //

  // const c = this;
  // const x = this.d3.scaleBand().rangeRound([0, this.width - this.margin.left - this.margin.right]).padding(0.1);
  // const y = this.d3.scaleLinear().rangeRound([this.height, 0]);
  //
  //   x.domain(this.data.map(function(d, i) { return c.xLabels[i]; }));
  //   y.domain([0, this.d3.max(this.data, function(d) { return d; })]);
  //
  // g.append('g').attr('class', 'axis axis--x')
  //   .attr('transform', 'translate(0,' + this.height + ')')
  //   .call(this.d3.axisBottom(x));
  //
  // g.append('g').attr('class', 'axis axis--y')
  //   .call(this.d3.axisLeft(y).ticks(6))
  //   .append('text')
  //   .attr('transform', 'rotate(-90)')
  //   .attr('y', 6)
  //   .attr('dy', '0.71em')
  //   .attr('text-anchor', 'end')
  //   .text('Type');
  //
  // console.log(' x (0) = ' + x(''));
  // let i = 0;
  //   g.selectAll('.bar')
  //     .data(this.data)
  //     .enter().append('rect')
  //     .attr('class', 'bar')
  //     .attr('x', d => x(this.xLabels[i++]))
  //     .attr('y', d => y(d))
  //     .attr('width', x.bandwidth())
  //     .attr('height', d => this.height - y(d));
  //
  //   i = 0;
  //   g.data(this.data).enter().append('text')
  //     .attr('x', (d) => {
  //       console.log('d = ' + d);
  //       return x.bandwidth() * i;
  //     })
  //     .attr('y', (d) => y(d))
  //     .attr('dy', '.35em')
  //     .text((d: number) => d);
  // }

}
