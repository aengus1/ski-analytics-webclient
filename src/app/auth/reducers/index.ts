import {ActionReducerMap, createFeatureSelector, createSelector} from '@ngrx/store';
import * as fromAuth from './auth.reducer';
import * as fromLoginPage from './login-page.reducer';
import * as fromSignupPage from './signup-page.reducer';
import * as fromForgotPage from './forgot-page.reducer';
import * as fromRoot from '../../reducers';

export interface AuthState {
  status: fromAuth.State;
  loginPage: fromLoginPage.State;
  signupPage: fromSignupPage.State;
  forgotPage: fromForgotPage.State;
}

export interface State extends fromRoot.State {
  auth: AuthState;
}

export const reducers: ActionReducerMap<AuthState> = {
  status: fromAuth.reducer,
  loginPage: fromLoginPage.reducer,
  signupPage: fromSignupPage.reducer,
  forgotPage: fromForgotPage.reducer
};

export const selectAuthState = createFeatureSelector<AuthState>('auth');

export const selectAuthStatusState = createSelector(
  selectAuthState,
  (state: AuthState) => state.status
);
export const getLoggedIn = createSelector(
  selectAuthStatusState,
  fromAuth.getLoggedIn
);

export const getUser = createSelector(selectAuthStatusState, fromAuth.getUser);

export const selectLoginPageState = createSelector(
  selectAuthState,
  (state: AuthState) => state.loginPage
);
export const getLoginPageError = createSelector(
  selectLoginPageState,
  fromLoginPage.getError
);
export const getLoginPagePending = createSelector(
  selectLoginPageState,
  fromLoginPage.getPending
);

export const selectSignupPageState = createSelector(
  selectAuthState,
  (state: AuthState) => state.signupPage
);
export const getSignupPageSignupError = createSelector(
  selectSignupPageState,
  fromSignupPage.getSignupError
);
export const getSignupPageStatus = createSelector(
  selectSignupPageState,
  fromSignupPage.getSignupStatus
);

export const getSignupPageConfirmError = createSelector(
  selectSignupPageState,
  fromSignupPage.getConfirmError
);

export const getSignupPageResendConfirmError = createSelector(
  selectSignupPageState,
  fromSignupPage.getResendConfirmError
);

export const selectForgotPageState = createSelector(
  selectAuthState,
  (state: AuthState) => state.forgotPage
);
export const getForgotPageError = createSelector(
  selectForgotPageState,
  fromForgotPage.getError
);
export const getForgotPagePending = createSelector(
  selectLoginPageState,
  fromForgotPage.getPending
);

