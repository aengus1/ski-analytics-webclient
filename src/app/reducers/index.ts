import {ActionReducer, ActionReducerMap, createFeatureSelector, createSelector, MetaReducer} from '@ngrx/store';
import {environment} from '../../environments/environment';
import * as fromLayout from '../shared/layout/reducers/layout.reducer';
import * as fromRouter from '@ngrx/router-store';
import {RouterStateUrl} from '../shared/utils';

export interface State {

  router: fromRouter.RouterReducerState<RouterStateUrl>;
  layout: fromLayout.State;
}



export const reducers: ActionReducerMap<State> = {
router: fromRouter.routerReducer,
  layout: fromLayout.reducer
};

// console.log all actions
export function logger(reducer: ActionReducer<State>): ActionReducer<State> {
  return function(state: State, action: any): State {
    console.log('state', state);
    console.log('action', action);

    return reducer(state, action);
  };
}


export const metaReducers: MetaReducer<State>[] = !environment.production ? [logger] : [];


/**
 * Layout Reducers
 */
export const getLayoutState = createFeatureSelector<fromLayout.State>('layout');

export const getNavbarCollapsed = createSelector(
  getLayoutState,
  fromLayout.getNavbarCollapsed
);


