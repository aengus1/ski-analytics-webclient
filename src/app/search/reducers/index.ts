import {ActionReducerMap, createFeatureSelector, createSelector} from '@ngrx/store';

import * as fromSearch from './search.reducer';
import * as fromRoot from '../../reducers';
import * as fromLayout from '../../shared/layout/reducers/layout.reducer';



export interface SearchState {
  search: fromSearch.State;
  error: any;
}
export const reducers: ActionReducerMap<SearchState> = {
  search: fromSearch.reducer,
  error: null
};


export interface State extends fromRoot.State {
  search: SearchState;
}


/**
 * The createFeatureSelector function selects a piece of state from the root of the state object.
 * This is used for selecting feature states that are loaded eagerly or lazily.
 */
export const getSearchState = createFeatureSelector<SearchState>('search');

export const getLayoutState = createFeatureSelector<fromLayout.State>('layout');

export const getSearchEntitiesState = createSelector(
  getSearchState,
  state => state.search
);

// export const selectSearchResults =  (state: SearchState) => state.search.searchResults;

export const selectSearchResults = createSelector(
  getSearchEntitiesState,
  state => state.searchResults
);




