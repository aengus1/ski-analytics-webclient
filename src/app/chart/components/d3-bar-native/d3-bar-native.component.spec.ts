import {async, ComponentFixture, TestBed} from '@angular/core/testing';

import {D3BarNativeComponent} from './d3-bar-native.component';
import {D3Service} from 'd3-ng2-service';
import {LoggerService} from '../../../shared/services/logger.service';
import {ChartModule} from '../../chart.module';
import {ConsoleLoggerService} from '../../../shared/services/console-logger.service';
import {Component, OnInit, ViewEncapsulation} from '@angular/core';
import {ChartOptions} from '../../ChartOptions';
import {ActivityModule} from '../../../activity/activity.module';
import {combineReducers, StoreModule} from '@ngrx/store';
import * as fromActivities from '../../../activity/reducers';
import {EffectsModule} from '@ngrx/effects';
import {RouterTestingModule} from '@angular/router/testing';
import {IntervalPipe} from '../../../shared/pipes/interval.pipe';

describe('D3BarNativeComponent', () => {
  let component: AppMockComponent;
  let fixture: ComponentFixture<AppMockComponent>;
  let compiled: HTMLElement | null;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      imports: [ChartModule],
      declarations: [AppMockComponent ],
      providers: [D3Service,
        { provide: LoggerService, useClass: ConsoleLoggerService },
        IntervalPipe]
    })
    .compileComponents();
  }));

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      imports: [ActivityModule,
        StoreModule.forRoot({
          activities: combineReducers(fromActivities.reducers),
        }),
        EffectsModule.forRoot([]),
        RouterTestingModule
      ],
      declarations: [AppMockComponent]
    });

    fixture = TestBed.createComponent(AppMockComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
    compiled = fixture.debugElement.nativeElement;
  }));

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('should match snapshot', () => {
    fixture.detectChanges();

    // noinspection TypeScriptUnresolvedFunction
    (<any>expect(fixture)).toMatchSnapshot();
  });
});

@Component({
  selector: 'app-mock-component',
  template: `<app-d3-bar-native
  [width]=layout.width
  [height]=layout.height
  [data] = layout.data
  [margin] = layout.margin
  [yLabel] = layout.yLabel
  [xLabels] = layout.xLabels
  [chartOptions] = layout.chartOptions>
    </app-d3-bar-native>`,
  encapsulation: ViewEncapsulation.None
})
class AppMockComponent  implements OnInit {


  public layout = {
    data: [10, 20, 30, 0, 40],
    width: 50,
    margin: 15,
    height: 50,
    yLabel: 'YLabelTest',
    xLabels: ['1', '2', '3', '4', '5'],
    chartOptions: new ChartOptions()
  };

  constructor() {

  }

  ngOnInit() {
    // run summarize activity manually as component expects activity to be summarized
    // ActivitySummaryService.summarizeActivity(this.layout.activity, null);
  }
}
