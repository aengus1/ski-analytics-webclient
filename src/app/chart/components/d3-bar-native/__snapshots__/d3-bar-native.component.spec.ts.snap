// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`D3BarNativeComponent should match snapshot 1`] = `
<app-mock-component
  layout={[Function Object]}
>
  <app-d3-bar-native
    ng-reflect-chart-options="[object Object]"
    ng-reflect-data="10,20,30,0,40"
    ng-reflect-height="50"
    ng-reflect-margin="15"
    ng-reflect-width="50"
    ng-reflect-x-labels="1,2,3,4,5"
    ng-reflect-y-label="YLabelTest"
  >
    <svg
      height="[height]"
      width="[width]"
    >
      <g
        transform="translate(15,15)"
      >
        <g
          class="axis--x"
          fill="none"
          font-family="sans-serif"
          font-size="10"
          style="text-anchor; start"
          text-anchor="middle"
          transform="translate(0,50)"
        >
          <path
            class="domain"
            d="M0,0H50"
            stroke="#000"
          />
        </g>
        <g
          class="bar-holder"
        >
          
          <rect
            class="bar"
            height="12.5"
            width="4"
            x="0"
            y="37.5"
          />
          <rect
            class="bar"
            height="25"
            width="4"
            x="10"
            y="25"
          />
          <rect
            class="bar"
            height="37.5"
            width="4"
            x="20"
            y="12.5"
          />
          <rect
            class="bar"
            height="0"
            width="4"
            x="30"
            y="50"
          />
          <rect
            class="bar"
            height="50"
            width="4"
            x="40"
            y="0"
          />
        </g>
        <g
          class="bar-text"
        >
          
          <text
            class="bar-text"
            height="12.5"
            style="display: inline;"
            width="4"
            x="0"
            y="37.5"
          >
             this.intervalPipe.transform(t) 
          </text>
          <text
            class="bar-text"
            height="25"
            style="display: inline;"
            width="4"
            x="10"
            y="25"
          >
             this.intervalPipe.transform(t) 
          </text>
          <text
            class="bar-text"
            height="37.5"
            style="display: inline;"
            width="4"
            x="20"
            y="12.5"
          >
             this.intervalPipe.transform(t) 
          </text>
          <text
            class="bar-text"
            height="0"
            style="display: inline;"
            width="4"
            x="30"
            y="50"
          >
             this.intervalPipe.transform(t) 
          </text>
          <text
            class="bar-text"
            height="50"
            style="display: inline;"
            width="4"
            x="40"
            y="0"
          >
             this.intervalPipe.transform(t) 
          </text>
        </g>
      </g>
    </svg>
  </app-d3-bar-native>
</app-mock-component>
`;
